cmd/note_common.go

```diff
@@ -85,10 +91,122 @@ func noteRunFn(cmd *cobra.Command, args []string) {
 		return
 	}
 
-	createNote(rn, isMR, int(idNum), msgs, filename, linebreak)
+	createNote(rn, isMR, int(idNum), msgs, filename, linebreak, commit)
+}
+
+func createCommitNote(rn string, mrID int, sha string, newFile string, oldFile string, oldline int, newline int, comment string, block bool) {
+	linetype := "old"
+	line := oldline
+	if oldline == -1 {
+		linetype = "new"
+		line = newline
+	}
+
+	if block {
+		webURL, err := lab.CreateMergeRequestCommitDiscussion(rn, mrID, sha, newFile, oldFile, line, linetype, comment)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(webURL)
+		return
+	}
+
+	webURL, err := lab.CreateCommitComment(rn, sha, newFile, oldFile, line, linetype, comment)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(webURL)
+}
+
+func getCommitBody(project string, commit string) (body string) {
+	//body is going to be the commit diff
+	ds, err := lab.GetCommitDiff(project, commit)
+	if err != nil {
+		fmt.Printf("    Could not get diff for commit %s.\n", commit)
+		return
+	}
+
+	if len(ds) == 0 {
+		log.Fatal("    No diff found for %s.", commit)
+	}
+
+	for _, d := range ds {
+		body = body + fmt.Sprintf("| newfile: %s oldfile: %s\n", d.NewPath, d.OldPath)
+		body = body + displayDiff(d.Diff, 0, 0, true)
+	}
+
+	return body
 }
 
-func createNote(rn string, isMR bool, idNum int, msgs []string, filename string, linebreak bool) {
+func createCommitComments(project string, mrID int, commit string, body string, block bool) {
+	// Go through the body line-by-line and find lines that do not
+	// begin with |.  These lines are comments that have been made
+	// on the patch.  The lines that begin with | contain patch
+	// tracking information (new line & old line number pairs,
+	// and file information)
+	scanner := bufio.NewScanner(strings.NewReader(body))
+	newfile := ""
+	oldfile := ""
+	oldLineNum := -1
+	newLineNum := -1
+	comments := ""
+	for scanner.Scan() {
+		if strings.HasPrefix(scanner.Text(), "| newfile:") {
+			if comments != "" {
+				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
+				comments = ""
+			}
+			// read filename
+			f := strings.Split(scanner.Text(), " ")
+			newfile = f[2]
+			if len(f) < 5 {
+				oldfile = ""
+			} else {
+				oldfile = f[4]
+			}
+		} else if strings.HasPrefix(scanner.Text(), "|") {
+			if comments != "" {
+				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
+				comments = ""
+			}
+			// read line numbers
+			fs := strings.Split(scanner.Text(), " ")
+
+			oldLineNum = -1
+			newLineNum = -1
+			for _, f := range fs {
+				if f == "" || f == "|" {
+					continue
+				}
+				val, err := strconv.Atoi(f)
+				if err != nil {
+					// NaN
+					if f == "@@" {
+						continue
+					}
```

Comment:
maybe f == "@@"  could be also checked above?
```suggestion
				if f == "" || f == "|" || f == "@@" {
					continue
				}
				val, err := strconv.Atoi(f)
				if err != nil {
					// NaN
```

Reply:
Good suggestion.  I've incorporated into v2.